src/plugins/analysis/strings/code/strings.py

```diff
@@ -2,54 +2,75 @@
 
 import re
 from re import Pattern
+from typing import TYPE_CHECKING, Iterable, List
+
+from pydantic import BaseModel, Field
+from semver import Version
 
 import config
-from analysis.PluginBase import AnalysisBasePlugin
+from analysis.plugin import AnalysisPluginV0
+from analysis.plugin.compat import AnalysisBasePluginAdapterMixin
+from plugins.analysis.strings.internal.string_eval import evaluate_string
 from plugins.mime_blacklists import MIME_BLACKLIST_COMPRESSED
 
+if TYPE_CHECKING:
+    from io import FileIO
+
+STRING_REGEXES = [
+    (b'[\x09-\x0d\x20-\x7e]{$len,}', 'utf-8'),
+    (b'(?:[\x09-\x0d\x20-\x7e]\x00){$len,}', 'utf-16'),
+]
+
 
-class AnalysisPlugin(AnalysisBasePlugin):
-    """
-    Extracts all printable Strings
-    """
+class StringMatch(BaseModel):
+    string: str
+    offset: int
+    relevance: float
 
-    NAME = 'printable_strings'
-    DEPENDENCIES = []  # noqa: RUF012
-    MIME_BLACKLIST = MIME_BLACKLIST_COMPRESSED
-    DESCRIPTION = 'extracts strings and their offsets from the files consisting of printable characters'
-    VERSION = '0.3.4'
-    FILE = __file__
 
-    STRING_REGEXES = [  # noqa: RUF012
-        (b'[\x09-\x0d\x20-\x7e]{$len,}', 'utf-8'),
-        (b'(?:[\x09-\x0d\x20-\x7e]\x00){$len,}', 'utf-16'),
-    ]
+class AnalysisPlugin(AnalysisPluginV0, AnalysisBasePluginAdapterMixin):
+    class Schema(BaseModel):
+        strings: List[StringMatch] = Field(description='An array of ASCII strings contained in this file.')
 
-    def additional_setup(self):
+    def __init__(self):
+        super().__init__(
+            metadata=(
+                self.MetaData(
+                    name='printable_strings',
+                    description='extracts strings and their offsets from the files consisting of printable characters',
+                    version=Version(1, 0, 0),
+                    mime_blacklist=MIME_BLACKLIST_COMPRESSED,
+                    Schema=self.Schema,
+                )
+            )
+        )
         self.regexes = self._compile_regexes()
 
     def _compile_regexes(self) -> list[tuple[Pattern[bytes], str]]:
         min_length = getattr(config.backend.plugin.get(self.NAME, {}), 'min-length', 8)
```

Comment:
This only works as long as we use `AnalysisBasePluginAdapterMixin`. 
Instead this should use `self.metadata.name`.

Reply:
Good catch. Totally overlooked that bit